core/src/main/java/nu/yona/server/device/service/DeviceService.java

```diff
@@ -156,18 +194,28 @@ private int findFirstFreeDeviceIndex(User userEntity)
 	}
 
 	@Transactional
-	public void deleteDevice(User userEntity, UserDevice device)
+	public void deleteDeviceAndNotifyBuddies(UUID userId, UUID deviceId)
 	{
-		Set<UserDevice> devices = userEntity.getDevices();
-		if (((devices.size() == 1) && devices.contains(device)))
-		{
-			throw DeviceServiceException.cannotDeleteLastDevice(device.getId());
-		}
-		deleteDeviceEvenIfLastOne(userEntity, device);
+		userService.updateUser(userId, userEntity -> {
+			UserDevice deviceEntity = getDeviceEntity(deviceId);
+			Set<UserDevice> devices = userEntity.getDevices();
+			if (((devices.size() == 1) && devices.contains(deviceEntity)))
+			{
+				throw DeviceServiceException.cannotDeleteLastDevice(deviceEntity.getId());
+			}
+
+			String oldName = deviceEntity.getName();
+			UUID deviceAnonymizedId = deviceEntity.getDeviceAnonymized().getId();
+
+			deleteDevice(userEntity, deviceEntity);
+
+			sendDeviceChangeMessageToBuddies(DeviceChange.DELETE, userEntity, Optional.of(oldName), Optional.empty(),
+					deviceAnonymizedId);
+		});
 	}
 
 	@Transactional
-	public void deleteDeviceEvenIfLastOne(User userEntity, UserDevice device)
+	public void deleteDevice(User userEntity, UserDevice device)
```

Comment:
`deleteDeviceWithoutNotifyingBuddies` or `deleteDeviceInternal` maybe works better?

Reply:
I just now see your comment. I made it ``deleteDeviceWithoutBuddyNotification``. It's not really internal, as it's a public method on the service, called during a migration step and during user deletion.